Restructured README file for Onboarding

schrpa opened the pull request:
---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [x] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
Moving the content of the README.md license section to the about section of the repository:
"License of our Project (mainly AGPL v3)"

This issue was discussed with Stefan and we decided to add the AGPL v3 license to the about section and remove it from the README.md.

### Tests - additional information
Not applicable

schrpa committed:
Display License in About Section

Moving the content of the README.md license section to the about section of the repository:
"License of our Project (mainly AGPL v3)"

schrpa committed:
Merge pull request #1 from schrpa/add-license-1

Display License in About Section

schrpa committed:
Added Onboarding Team Introduction

Changed the file as discussed in the prototype.

schrpa committed:
Create onboarding.md

This file is needed for the link in the README file and will contain all necessary information for newcomers in the future.

schrpa committed:
Added Link to Onboarding File

dmetzner commented on .github/onboarding.md:
```diff
@@ -0,0 +1 @@
+👷 **__We are currently working on an improved onboarding experience!__** 👷
```
why in the .github folder?

dmetzner replied:
If we are thinking about moving from the wiki to the "code" in the repository, we might should create a documentation directory?

schrpa replied:
Short Answer:
To me, it seems to be the logically correct position based on the location of the contribution.md file.

Long Answer:
As discussed with the coordinators of all teams, the plan is to automate the onboarding process. For this purpose, we agreed to use markdown files on Github. The core of the onboarding process will be introducing the workflow at Catrobat's development teams. This workflow is already (partly) described in the contribution.md file at Catroweb and will be adapted to fit the needs of newcomers and current contributors. It will serve as a step-by-step guide through the development process of a single ticket. Additional information that is only relevant to the newcomers will be put into the onboarding.md file to serve as a source of information linking to existing documentation (e.g., wiki, confluence) and the contribution.md file. The contribution.md file is correctly placed in the .github, and used in https://github.com/Catrobat/Catroweb/community. Creating a documentation directory for a single file (onboarding.md) is a bit overhead in this case.

dmetzner replied:
looked it up. GitHub looks for files such as the contribution.md in the root, docs and .github directory. Since onboarding.md as far as i know is no file that is supported by github, the docs directory would make more sense to me. Of course, that's no game breaker and just a matter of preference, so we can keep it that way. Was just interested in your intentions :)

dmetzner commented on README.md:
```diff
@@ -1,49 +1,55 @@
+[![CSS](https://img.shields.io/badge/CSS-darkblue)](https://www.w3schools.com/css/)
+[![PHP](https://img.shields.io/badge/PHP-with_Symfony_and_Laravel-green?labelColor=darkblue)](https://www.php.net/)
+[![Symfony](https://img.shields.io/badge/Symfony-darkblue)](https://symfony.com/)
+[![Laravel](https://img.shields.io/badge/Laravel-darkblue)](https://laravel.com/)
```
We don't use laravel in this project

dmetzner commented on README.md:
```diff
@@ -1,49 +1,55 @@
+[![HTML](http://img.shields.io/badge/HTML-darkblue)](https://www.w3schools.com/html/)
+[![CSS](https://img.shields.io/badge/CSS-darkblue)](https://www.w3schools.com/css/)
+[![PHP](https://img.shields.io/badge/PHP-with_Symfony_and_Laravel-green?labelColor=darkblue)](https://www.php.net/)
+[![Symfony](https://img.shields.io/badge/Symfony-darkblue)](https://symfony.com/)
```
Symfony is a framework, no language.

dmetzner commented on README.md:
```diff
@@ -1,49 +1,55 @@
-## How to get started: 
+### Languages
+[![HTML](http://img.shields.io/badge/HTML-darkblue)](https://www.w3schools.com/html/)
+[![CSS](https://img.shields.io/badge/CSS-darkblue)](https://www.w3schools.com/css/)
```
To be more specific we use sass

dmetzner commented on README.md:
```diff
@@ -1,49 +1,55 @@
 
-You can find all important HowTo's in our [wiki](https://github.com/Catrobat/Catroweb-Symfony/wiki) and our [contribution guidelines](https://github.com/Catrobat/Catroweb/blob/develop/.github/contributing.md).
-Make sure all tests are passing when creating a pull request,
+### Infrastructure (IDE, Version Control, Issue Tracking, ...)
```
GitHub Actions would fit good into this context

dmetzner replied:
If we mention tools such as the IDE, which is, by all means, optional, why remove tools that must be used such as webpack, phpstan, psalm, ... from the tech stack?

dmetzner commented on README.md:
```diff
@@ -1,49 +1,55 @@
+### Communication
+[![Slack](https://img.shields.io/badge/Slack-green)](https://slack.com/)
+[![Discord](https://img.shields.io/badge/Discord-green)](https://discord.com/)
 
```
also directly in GitHub via PRs

dmetzner commented on README.md:
```diff
@@ -1,49 +1,55 @@
+[![PHP](https://img.shields.io/badge/PHP-with_Symfony_and_Laravel-green?labelColor=darkblue)](https://www.php.net/)
+[![Symfony](https://img.shields.io/badge/Symfony-darkblue)](https://symfony.com/)
+[![Laravel](https://img.shields.io/badge/Laravel-darkblue)](https://laravel.com/)
+[![JavaScript](https://img.shields.io/badge/JavaScript-darkblue)](https://www.w3schools.com/js/)
```
SQL should also be known to the backend devs.

dmetzner commented on README.md:
```diff
@@ -1,49 +1,55 @@
-- specification: http://developer.catrobat.org/Catroweb-API
-- repository: https://github.com/Catrobat/Catroweb-API
+## Technologies
+Below you can find an outline of the technologies we currently use at Catroweb. If you want to contribute to our team, your tech stack should cover most of the technologies. But don't worry! If you are uncertain about a specific technology, you can learn it along the way. Google is your friend ;)
```
Should we really mention Google here? @HCrane

dmetzner commented on README.md:
```diff
@@ -1,49 +1,55 @@
-The website features Catrobat projects (also called Catrobat programs) created with Pocket Code. 
-Uploaded projects can be shared and downloaded. Additionally, the platform provides users with features typical for a social network.
+# Catroweb
+Catroweb is the name of the "Pocket Code" [sharing platform](https://share.catrob.at/app/), where our community can upload their projects. If you uploaded your game in the previous step, it should already be visible to all other users. They can now download and remix your masterpiece. Additionally, the platform provides users with features typical for a social network.
```
game is only a subset of the possible projects. Should we really reduce the set to games here?

dmetzner commented on README.md:
````diff
@@ -1,49 +1,55 @@
-[![codecov](https://codecov.io/gh/Catrobat/Catroweb/branch/develop/graph/badge.svg)](https://codecov.io/gh/Catrobat/Catroweb)
+⚠️ **__If you have not heard about who Catrobat is, what we are doing and in what ways you can contribute, then you might want to take a step back and check out the start of our [step-by-step guide](https://github.com/Catrobat/Catroid/blob/develop/README.md).__**
+```diff
+👷 TODO: Change link to README in Catrobat's .github repository when added!
````
Wouldn't it be better to create a ticket instead of this ToDo here?

schrpa committed:
Update README.md

Changes related to the review of https://github.com/Catrobat/Catroweb/pull/3155

schrpa committed:
Update README.md

Changed Symfony label color

schrpa committed:
Update README.md

Changed link in "start contribution" section to point to Catroweb repository of Catrobat.

mfuerni commented on README.md:
```diff
@@ -1,49 +1,55 @@
-[License](https://catrob.at/licenses) of our project (mainly AGPL v3).
+## Issues
+You found a bug? 
+1. Perfect, please check the [open issues on Jira](https://jira.catrob.at/browse/SHARE-529?jql=project%20%3D%20%22SHARE%22%20AND%20issuetype%20%3D%20Bug%20AND%20resolution%20%3D%20Unresolved%20ORDER%20BY%20priority%20DESC%2C%20key%20DESC) first to avoid duplicate tickets!
```
Correct Jira URL: https://jira.catrob.at/projects/SHARE/issues/?filter=allopenissues
```suggestion
1. Perfect, please check the [open issues on Jira](https://jira.catrob.at/projects/SHARE/issues/?filter=allopenissues) first to avoid duplicate tickets!
```

schrpa committed:
Update README.md

Changed all issues on Jira filter

Co-authored-by: Michael Fürnschuß <mfuerni@users.noreply.github.com>

schrpa committed:
Update Onboarding File
